src/client/components/app/Transactions/DepositTx.tsx

```diff
@@ -109,6 +106,21 @@ export const DepositTx: FC<DepositTxProps> = ({
     };
   }, []);
 
+  useEffect(() => {
+    if (!selectedVault || !selectedSellTokenAddress || !walletIsConnected) return;
+    const fetchAllowance = async () => {
+      setIsFetchingAllowance(true);
+      await dispatch(
+        VaultsActions.getVaultAllowance({
+          vaultAddress: selectedVault.address,
+          tokenAddress: selectedSellTokenAddress,
+        })
+      );
+      setIsFetchingAllowance(false);
+    };
+    fetchAllowance();
+  }, [selectedSellTokenAddress, selectedVault?.address, walletIsConnected]);
```

Comment:
Would it be possible to extract just this bit to its own hook so we can reuse in `LabDepositTx.tsx` -- if too hard let's just leave as is

Reply:
Yeah, it would be ideal. Should we refactor in a new PR, so we release a fix for this bug that is in prod now?

Reply:
Sounds good